src/_base/helm/app/templates/application/image-pull-config.yaml

```diff
@@ -9,8 +9,9 @@ kind: Secret
 metadata:
   name: {{ .Values.resourcePrefix }}image-pull-config
   annotations:
-    sealedsecrets.bitnami.com/cluster-wide: "true"
+    argocd.argoproj.io/sync-wave: "4"
```

Comment:
documentation says default is 0 and lower number waves are done before higher, so this would already be the case, right, unless using nevative number waves?

it's useful for everything that has no dependency on other resources to be in the same wave, so I'm not sure but the default might be better?

Reply:
what you might have been finding is the secret wasn't decrypted in time in a real run maybe? Increasing the wave to make it happen later only increases that

Reply:
For whatever reason, one of our environments doing this conversion did not put the sealed secret for image pull config on the cluster before trying to do app-init.

It's true that the default of 0 should have been sufficient, but let's be explicit about the ordering.

Reply:
is every resource like this then? should we set it to 0 for explicitness?

Reply:
If and when we spot a problem, can change it :+1:

Reply:
likewise app-init will have a need for the environment secrets